src/System.Text.Encodings.Web/tests/JavaScriptStringEncoderTests.Relaxed.cs

```diff
@@ -0,0 +1,318 @@
+
+                        if (i <= 0x001F || (0x007F <= i && i <= 0x9F))
+                        {
+                            mustEncode = true; // control char
+                        }
+                        else if (!UnicodeHelpers.IsCharacterDefined((char)i))
+                        {
+                            mustEncode = true; // undefined (or otherwise disallowed) char
+                        }
+
+                        if (mustEncode)
+                        {
+                            expected = string.Format(CultureInfo.InvariantCulture, @"\u{0:X4}", i);
+                        }
+                        else
+                        {
+                            expected = input; // no encoding
+                        }
+                    }
+                }
+
+                string retVal = encoder.JavaScriptStringEncode(input);
+                Assert.Equal(expected, retVal);
+            }
+
+            // Act & assert - astral chars
+            for (int i = 0x10000; i <= 0x10FFFF; i++)
+            {
+                string input = char.ConvertFromUtf32(i);
+                string expected = string.Format(CultureInfo.InvariantCulture, @"\u{0:X4}\u{1:X4}", (uint)input[0], (uint)input[1]);
+                string retVal = encoder.JavaScriptStringEncode(input);
+                Assert.Equal(expected, retVal);
+            }
+        }
+
+        [Fact]
+        public void JavaScriptStringEncode_BadSurrogates_ReturnsUnicodeReplacementChar_Relaxed()
+        {
+            // Arrange
+            JavaScriptStringEncoder encoder = JavaScriptStringEncoder.UnsafeRelaxedJsonEscaping; // allow all codepoints
+
+            // "a<unpaired leading>b<unpaired trailing>c<trailing before leading>d<unpaired trailing><valid>e<high at end of string>"
+            const string input = "a\uD800b\uDFFFc\uDFFF\uD800d\uDFFF\uD800\uDFFFe\uD800";
+            const string expected = "a\uFFFDb\uFFFDc\uFFFD\uFFFDd\uFFFD\\uD800\\uDFFFe\uFFFD"; // 'D800' 'DFFF' was preserved since it's valid
+
+            // Act
+            string retVal = encoder.JavaScriptStringEncode(input);
+
+            // Assert
+            Assert.Equal(expected, retVal);
+        }
+
+        [Fact]
+        public void JavaScriptStringEncode_EmptyStringInput_ReturnsEmptyString_Relaxed()
+        {
+            // Arrange
+            JavaScriptStringEncoder encoder = JavaScriptStringEncoder.UnsafeRelaxedJsonEscaping;
+
+            // Act & assert
+            Assert.Equal("", encoder.JavaScriptStringEncode(""));
+        }
+
+        [Fact]
+        public void JavaScriptStringEncode_InputDoesNotRequireEncoding_ReturnsOriginalStringInstance_Relaxed()
+        {
+            // Arrange
+            JavaScriptStringEncoder encoder = JavaScriptStringEncoder.UnsafeRelaxedJsonEscaping;
+            string input = "Hello, there!";
+
+            // Act & assert
+            Assert.Same(input, encoder.JavaScriptStringEncode(input));
```

Comment:
What are these comments saying?  ("Arrange", "Act & assert")

Reply:
Followed existing pattern. I have seen the "Arrange, "Act", "Assert" pattern used elsewhere in unit tests so I retained them: https://docs.microsoft.com/en-us/visualstudio/test/unit-test-basics?view=vs-2019#write-your-tests